### PR TITLE
feat: YankWord task

### PR DESCRIPTION
**Pull request by mertovun**
* Minor refactoring on utility functions, and addition of random word selection/highlighting utilities

* YankWord task implemented with the description "Yank the highlighted text" and added to the task index.

* The old YankWordTask implementation has been saved to another file for reference and further discussion.

### CODE REVIEW

**Comment by Weyaaron** [comment]
I think some of the additions to the utility class are neat and will be merged eventually I suppose. But the way your new method 'create highligt'(See code comment on this pr) works is not good coding policy. A function should do one thing and one thing only. A function that creates a highlight should not do the unrelated side effect of returning something unrelated to the highlight. 

My suggestion is that I leave this open and maybe my suggestions could be implemented on top? If not, you might be better off splitting this pr into the utility and the task part. I have no particular preference.

File: lua/nvim-training/tasks/yank_word.lua [hunk 1]
```diff
@@ -1,67 +1,26 @@
...
+	setmetatable(base, { __index = YankWord })
 	self.autocmd = "TextYankPost"
 	self.target_text = ""
 	self.jump_distance = 2
 
 	local function _inner_update()
-		self.custom_lorem_ipsum = string.gsub(utility.lorem_ipsum_lines(), ",", "")
-		self.custom_lorem_ipsum = string.gsub(self.custom_lorem_ipsum, "%.", "")
-		utility.update_buffer_respecting_header(self.custom_lorem_ipsum)
+		utility.set_buffer_to_lorem_ipsum_and_place_cursor_randomly()
```
Comment (Weyaaron) [inline lua/nvim-training/tasks/yank_word.lua:15]: This line is unexpected, and code should not surprise the readers. This line does.

File: lua/nvim-training/utility.lua [hunk 2]
```diff
@@ -7,25 +7,51 @@ function utility.set_buffer_to_lorem_ipsum_and_place_cursor_randomly()
...
 function utility.move_cursor_to_random_point()
 	vim.api.nvim_win_set_cursor(0, utility.calculate_random_point_in_text_bounds())
 end
 
+function utility.highlight_random_word()
+	local x = utility.random_line_index()
+	local line = utility.get_line(x)
+	local wordParams = utility.get_word_params(line)
+	local selected = wordParams[math.random(1, #wordParams)]
+	return utility.create_highlight(x, selected[1]-1, selected[2])
```
Comment (Weyaaron) [inline lua/nvim-training/utility.lua]: This function should not have a return value, especially not something unrelated to its tasks. Maybe just remove it? But then you have to rework the yank task accordingly.

**Comment by Weyaaron** [comment]
And I am unsure if the movement belongs to the task at all. Maybe its best to stick to the word at the cursor instead of requiring some additional steps.

File: lua/nvim-training/tasks/yank_word.lua (hunk 1 above)
Comment (Weyaaron) [reply lua/nvim-training/tasks/yank_word.lua:15]: Hm, the placement of my comment is somewhat off, the comment is about the line where you set target text.

**Comment by mertovun** [comment]
> And I am unsure if the movement belongs to the task at all. Maybe its best to stick to the word at the cursor instead of requiring some additional steps.

I see your point. The extension could offer both granular tasks for individual commands and composite tasks for more realistic editing scenarios. I believe navigating to a word and copy it is a fundamental operation in the daily use and lies somewhere in between. We could consider renaming the current task to better reflect its intermediate nature, to address your concerns.

**Comment by Weyaaron** [comment]
I have opened an issue on this, we should discuss it over there. I will merge this particular instance  and go from there, I appreciate the contribution and the discussion.

**Comment by Weyaaron** [comment]
And I will take care of the renaming and might make some slight adjustments for the pr into main.

**Comment by Weyaaron** [comment]
There is some code left over that does not align with the desired changes, I will do this as well, I consider this a minor issue.

**Comment by Weyaaron** [comment]
> There is some code left over that does not align with the desired changes, I will do this as well, I consider this a minor issue.

These turned out to be not that minor, but it helped me with interacting with your code. All should be done now, if you have concerns let me know.

**Pull request merged**